site-src/geps/gep-1742.md

```diff
@@ -0,0 +1,315 @@
+# GEP-1742: Timeouts
+
+* Issue: [#1742](https://github.com/kubernetes-sigs/gateway-api/issues/1742)
+* Status: Provisional
+
+(See status definitions [here](overview.md#status).)
+
+## TLDR
+
+Create some sort of design so that Gateway API objects can be used to configure
+timeouts for different types of connection.
+
+## Goals
+
+- Create some method to configure some timeouts
+- Timeout config must be applicable to most if not all Gateway API implementations.
```

Comment:
It would be good to capture what types of timeouts are relevant to a broad set of users even if implementations currently don't support it.

Then we have enough data to say to implementors - 'if you add support for X,Y timeouts you'll handle 80% of use cases'